src/vs/workbench/contrib/chat/browser/chat.contribution.ts

```diff
@@ -212,11 +212,18 @@ configurationRegistry.registerConfiguration({
 		},
 		'chat.tools.autoApprove': {
 			default: false,
-			description: nls.localize('chat.tools.autoApprove', "Controls whether tool use should be automatically approved ('YOLO mode'). Can be set to `true`, or an array of tool names to automatically approve."),
+			description: nls.localize('chat.tools.autoApprove', "Controls whether tool use should be automatically approved ('YOLO mode'). Can be set to `true`, or a mapping of tool names to automatically approve."),
 			oneOf: [
 				{ type: 'boolean' },
-				{ type: 'array', items: { type: 'string' } }
+				{ type: 'object', additionalProperties: { type: 'boolean' } }
```

Comment:
Going to quickly validate that the policy file generation plays nice with a setting with this kind of type structure

Reply:
The `config.type` here ends up being undefined, so as-is this policy will not get registered

<img width="1254" alt="image" src="https://github.com/user-attachments/assets/87ebedb0-8b04-4f66-a03b-724b7f9eee97" />

https://github.com/microsoft/vscode/blob/f1c1b5266a66c2192d42db02b1db6b5da14bb52b/src/vs/platform/configuration/common/configurations.ts#L135-L136

Reply:
cc @sandy081

Reply:
Yes, unfortunately the policy parsing supports only single type.

CC @joaomoreno

Reply:
That's unfortunate. If we keep this as a string array, can the policy parser deal with that?

Reply:
Yea, this works nicely

```ts
		'chat.tools.autoApprove': {
			default: [],
			description: nls.localize('chat.tools.autoApprove', "Controls whether tool use should be automatically approved ('YOLO mode'). Can be set to `true`, or a mapping of tool names to automatically approve."),
			type: 'array',
			tags: ['preview'],
			policy: {
				name: 'ChatToolsAutoApprove',
				minimumVersion: '1.99',
				previewFeature: true,
				defaultValue: "[]"
			},
			items: {
				type: 'string',
			},
		},
```
<img width="982" alt="image" src="https://github.com/user-attachments/assets/6962dd8d-1a0c-49dc-8341-c3187c2764e4" />

<img width="725" alt="image" src="https://github.com/user-attachments/assets/565fcb4b-b98b-40e9-980c-242fff0db001" />

Reply:
Talking to Kai, we'll probably keep this as a boolean for this release and expand it later.